src/components/CardList.tsx

```diff
@@ -71,7 +71,36 @@ const Image = styled(GatsbyImage)`
   margin-top: 4px;
 `
 
-const CardList = ({ content, className, clickHandler }) => (
+type ForbidOptional<T = {}> = {
+  [P in keyof T]?: never
+}
+
+type OptionalCardImageProp = {
+  image: string
+  alt: string
+}
+
+type CardImage = ForbidOptional<OptionalCardImageProp>
+
+export type CardListItem = {
+  title: string
+  description?: string
+  caption?: string
+  link?: string
+  id?: string | number
+} & (OptionalCardImageProp | CardImage)
```

Comment:
Cool! Lets move these types to `src/types.ts` to reuse them in other components.

I've also made a few name adjustments.

```ts
// src/types.ts

export type ForbidOptional<T = {}> = {
  [P in keyof T]?: never
}

type OptionalImageProp = {
  image: string
  alt: string
}

type ForbidOptionalImageProp = ForbidOptional<ImageProp>

export type ImageProp = OptionalImageProp | ForbidOptionalImageProp
```

Reply:
Done :)

Reply:
Done , send me money or Bitcoin I love you